mne/tests/test_epochs.py

```diff
@@ -4795,3 +4795,45 @@ def test_epochs_saving_with_annotations(tmp_path):
     loaded_epochs = read_epochs(fname)
     assert epochs._raw_sfreq == loaded_epochs._raw_sfreq
     assert loaded_epochs.annotations is None
+
+
+def _get_empty_parametrize():
+    test_methods = {
+        "add_reference_channels": {"ref_channels": "EEG 999"},
+        "apply_function": {"fun": lambda x: x},
+        "apply_hilbert": {},
+        "as_type": {},
+        "average": {},
+        "compute_psd": {},
+        "drop_channels": {"ch_names": ["EEG 014"]},
+        "filter": {"l_freq": 1, "h_freq": 40},
+        "interpolate_bads": {},
+        "pick": {"picks": [0]},
+        "pick_channels": {"ch_names": ["EEG 014"]},
+        "pick_types": {"eeg": True},
+        "plot": {},
+        "plot_image": {},
+        "plot_psd": {},
+        "plot_psd_topo": {"tmin": 0.1, "tmax": 0.2},
+        "plot_psd_topomap": {},
+        "plot_topo_image": {},
+        "resample": {"sfreq": 100},
+        "reorder_channels": {"ch_names": ["EEG 014"]},
+        "savgol_filter": {"h_freq": 40},
+        "set_eeg_reference": {},
+        "shift_time": {"tshift": 0.1},
+        "standard_error": {},
+        "to_data_frame": {},
+    }
+    arg_values = [(k, v) for k, v in test_methods.items()]
+    arg_ids = test_methods.keys()
+    return {"argnames": "method", "argvalues": arg_values, "ids": arg_ids}
+
+
+@pytest.mark.parametrize(**_get_empty_parametrize())
+def test_empty_error(method, epochs_empty):
+    """Test that a helpful error is raised when certain methods are called."""
+    # Test that a RuntimeError is raised when certain methods are called
+    with pytest.warns(RuntimeWarning, match="dropped"):
```

Comment:
This is probably where I get something wrong. It just doesn't seem to be catching the `RuntimeWarning `from `mne.utils.warn`